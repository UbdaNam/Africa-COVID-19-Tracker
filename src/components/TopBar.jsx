import '../styles/TopBar.css';
import PropTypes from 'prop-types';

export default function TopBar({ src, title, population }) {
  return (
    <div className="content-wrapper">
      <div className="img-container">
        <img src={src} alt={title} />
      </div>
      <div className="content">
        <h2>{title}</h2>
        <p>
          population:
          {population}
        </p>
      </div>
    </div>
  );
}

TopBar.propTypes = {
  src: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  population: PropTypes.string.isRequired,
};
