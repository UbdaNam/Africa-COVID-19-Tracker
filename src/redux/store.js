import { configureStore } from "@reduxjs/toolkit";
import countriesReducer from "./countries/countriesSlice";
import detailsReducer from "./details/detailsSlice";

const store = configureStore({
  reducer: {
    countries: countriesReducer,
    countryDetails: detailsReducer,
  },
});

export default store;
