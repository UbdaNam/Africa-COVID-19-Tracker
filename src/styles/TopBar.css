.content-wrapper {
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  padding: 1rem;
  margin-top: 56.8px;
  background-color: #5688e4;
}

.img-container {
  display: flex;
  align-items: center;
  max-width: 30%;
  max-height: 250px;
  overflow: hidden;
}

.img-container img {
  width: 100%;
  height: 100%;
  object-fit: contain;
}

.content h2 {
  color: #fafafa;
}

.content p {
  color: #fafafa;
}
