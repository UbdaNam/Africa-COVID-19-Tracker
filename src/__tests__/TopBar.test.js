import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import store from '../redux/store';
import TopBar from '../components/TopBar';

const MockProps = {
  src: 'https://disease.sh/assets/img/flags/et.png',
  title: 'Ethiopia',
  population: '23458262',
};

const TopBarSnap = renderer.create(
  <Provider store={store}>
    <Router>
      <TopBar
        src={MockProps.src}
        title={MockProps.title}
        population={MockProps.population}
      />
    </Router>
  </Provider>,
);

describe('Top bar component render correctly', () => {
  test('Check if top bar renders correctly', () => {
    expect(TopBarSnap).toMatchSnapshot();
  });

  test("Check if alt text 'Ethiopia' exist inside the DOM", () => {
    render(
      <Provider store={store}>
        <Router>
          <TopBar
            src={MockProps.src}
            title={MockProps.title}
            population={MockProps.population}
          />
          {' '}
        </Router>
      </Provider>,
    );

    const image = screen.getByAltText('Ethiopia');

    expect(image.src).toContain('https://disease.sh/assets/img/flags/et.png');
    expect(screen.getByAltText('Ethiopia')).toBeInTheDocument();
  });
});
