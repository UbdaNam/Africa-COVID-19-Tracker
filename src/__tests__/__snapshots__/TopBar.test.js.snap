// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top bar component render correctly Check if top bar renders correctly 1`] = `
<div
  className="content-wrapper"
>
  <div
    className="img-container"
  >
    <img
      alt="Ethiopia"
      src="https://disease.sh/assets/img/flags/et.png"
    />
  </div>
  <div
    className="content"
  >
    <h2>
      Ethiopia
    </h2>
    <p>
      population:
      23458262
    </p>
  </div>
</div>
`;
